src/runtime_src/core/pcie/linux/shim.cpp

```diff
@@ -2214,6 +2214,15 @@ register_xclbin(const xrt::xclbin&)
   throw xrt_core::ishim::not_supported_error{__func__};
 }
 
+//Exec Buf with ctx handle.
+void
+shim::
+exec_buf_ctx(xclBufferHandle boh, const xrt::hw_context& hwctx)
+{
+  //TODO: Implement new fuction
+  throw xrt_core::ishim::not_supported_error{ __func__ };
```

Comment:
If we merge this PR, then nothing will work.  For now you should simply call exec_buf(boh).
```suggestion
  // TODO: Implement new function, for now just call legacy exec_buf.
  exec_buf(boh);
```

Reply:
yes @stsoe , as we removed try catch in kds_device.exec_buf() in hw_queue.cpp. This make sense.